activerecord/lib/active_record/relation.rb

```diff
@@ -675,7 +675,13 @@ def values
     end
 
     def inspect
-      entries = records.take([limit_value, 11].compact.min).map!(&:inspect)
+      num_to_take = [limit_value, 11].compact.min
+      if values.any?
```

Comment:
Good question! What you see here is the final rewrite after breaking and then fixing `AssociationProxyTest#test_inspect_does_not_reload_a_not_yet_loaded_target`. Basically, if I understand correctly, if a `has_many` association is built with initial values, it is not `loaded?`, in which case loading (by calling `limit(...).to_a`) would discard those initial values.

Thanks for prompting me to do this, I will add a comment with a brief explanation.